packages/redis/src/RedisStreamSink.ts

```diff
@@ -0,0 +1,96 @@
+import {
+    IOutputSink,
+    IPublishedMessage,
+    IOutputSinkGuarantees,
+    IRequireInitialization,
+    IDisposable,
+    IComponentContext,
+    makeLifecycle,
+    Lifecycle,
+    IMetrics,
+    failSpan,
+    DefaultComponentContext,
+    OutputSinkConsistencyLevel,
+} from "@walmartlabs/cookie-cutter-core";
+import { Span, Tags, Tracer } from "opentracing";
+
+import { IRedisInputStreamOptions, redisClient, IRedisClient, RedisMetadata } from ".";
+import { RedisOpenTracingTagKeys, RedisMetrics, RedisMetricResults } from "./RedisClient";
+
+export class RedisStreamSink
+    implements IOutputSink<IPublishedMessage>, IRequireInitialization, IDisposable {
+    public guarantees: IOutputSinkGuarantees;
+    private client: Lifecycle<IRedisClient>;
+    private tracer: Tracer;
+    private metrics: IMetrics;
+    private spanOperationName: string = "Redis Output Sink Client Call";
+
+    constructor(private readonly config: IRedisInputStreamOptions) {
+        this.tracer = DefaultComponentContext.tracer;
+        this.metrics = DefaultComponentContext.metrics;
+        this.guarantees = {
+            consistency: OutputSinkConsistencyLevel.None,
```

Comment:
it seems like the XAdd command supports multiple items at once. could we use that to make this sink atomic?